### PR TITLE
#889 Updated documentation and added test cases for derive_vars_duration() and compute_duration()

### PR DESCRIPTION
**Pull request by SyedMubasheer**
…on() and compute_duration()

Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: R/compute_duration.R [hunk 1]
```diff
@@ -96,6 +100,22 @@
 #'   out_unit = "years",
 #'   add_one = FALSE
 #' )
+#'
+#' # derive duration in hours
```
Comment (bms63) [inline R/compute_duration.R:104]:
```suggestion
#' # Derive duration in hours
```

File: R/compute_duration.R [hunk 2]
```diff
@@ -96,6 +100,22 @@
...
+#'compute_duration(
+#'   start_date = lubridate::ymd_hms("2020-12-06T9:00:00"),
+#'   end_date = lubridate::ymd_hms("2020-12-06T13:30:00"),
+#'   out_unit = "hours",
+#'   floor_in = FALSE,
+#'   add_one = FALSE,
+#' )
+#'
+#' # derive duration in days after imputation
```
Comment (bms63) [inline R/compute_duration.R:113]:
```suggestion
#' # Derive duration in days after imputation
```

File: R/derive_vars_duration.R [hunk 3]
```diff
@@ -89,6 +89,7 @@
 #' library(lubridate)
 #' library(tibble)
 #'
+#' # derive age in years
```
Comment (bms63) [inline R/derive_vars_duration.R:92]:
```suggestion
#' # Derive age in years
```

File: R/derive_vars_duration.R [hunk 4]
```diff
@@ -106,6 +107,42 @@
 #'   add_one = FALSE,
 #'   trunc_out = TRUE
 #' )
+#'
+#' # derive adverse event duration in days
```
Comment (bms63) [inline R/derive_vars_duration.R:111]:
```suggestion
#' # Derive adverse event duration in days
```

File: R/derive_vars_duration.R [hunk 5]
```diff
@@ -106,6 +107,42 @@
...
+#'
+#' derive_vars_duration(data,
+#'   new_var = ADURN,
+#'   new_var_unit = ADURU,
+#'   start_date = ASTDT,
+#'   end_date = AENDT,
+#'   out_unit = "days"
+#' )
+#'
+#' # derive adverse event duration in minutes
```
Comment (bms63) [inline R/derive_vars_duration.R:128]:
```suggestion
#' # Derive adverse event duration in minutes
```

File: tests/testthat/test-compute_duration.R [hunk 6]
```diff
@@ -58,3 +58,28 @@ test_that("age in weeks", {
...
+  expect_equal(
+    compute_duration(
+      convert_dtc_to_dt("2020-12", date_imputation = "first"),
+      ymd("2020-12-10"),
+      out_unit = "days"
+    ),
+    10
+  )
+})
+
```
Comment (bms63) [inline tests/testthat/test-compute_duration.R:85]: Can you look at the Unit Test Guidance to see how to write the tests?  image

Please renumber the tests as appropriate.

**Comment by bms63** [comment]
Many Thanks Syed!!  This is your first PR for admiral correct!?!  I'm excited for you!

Few things:

Can you fix the failing lintr checks and the R-CMD checks?  It looks like the unit tests are failing and there are superfluous lines in the test code and that is where the lint is failing.

You can use `lintr::lint_package()` to get exact place where the issues are occurring.

Also start comments in proper case.  

Please go through checklist as well.

**Comment by SyedMubasheer** [comment]
> Many Thanks Syed!! This is your first PR for admiral correct!?! I'm excited for you!
> 
> Few things:
> 
> Can you fix the failing lintr checks and the R-CMD checks? It looks like the unit tests are failing and there are superfluous lines in the test code and that is where the lint is failing.
> 
> You can use `lintr::lint_package()` to get exact place where the issues are occurring.
> 
> Also start comments in proper case.
> 
> Please go through checklist as well.

Thanks for reviewing @bms63!! Yes It's my first PR :) Sure will update accordingly.

**Comment by bms63** [comment]
@SyedMubasheer I commented out the failing test and it passes the two CMD checks but not 3.6 as another test fails there!  Not exactly sure what is going on here.  @thomas-neitmann any ideas?

**Comment by thomas-neitmann** [comment]
@SyedMubasheer Please removed the commented test case to get the lintr workflow running without errors. 
image

**Comment by bms63** [comment]
image

So this is a big red flag for me!  Please be careful what you committ

**Review by bms63 — APPROVED** [review]
Many thanks Syed!

**Comment by SyedMubasheer** [comment]
> Many thanks Syed!

Thank you @bms63  for the support and making me hit my first PR successfully :)!!

**Pull request merged**